executor/common_linux.h

```diff
@@ -2860,16 +2860,16 @@ static long syz_read_part_table(volatile unsigned long size, volatile unsigned l
 #include <string.h>
 #include <sys/mount.h>
 
-// syz_mount_image(fs ptr[in, string[disk_filesystems]], dir ptr[in, filename], size intptr, nsegs len[segments], segments ptr[in, array[fs_image_segment]], flags flags[mount_flags], opts ptr[in, fs_options[vfat_options]]) fd_dir
+// syz_mount_image(fs ptr[in, string[disk_filesystems]], dir ptr[in, filename], size intptr, nsegs len[segments], segments ptr[in, array[fs_image_segment]], flags flags[mount_flags], opts ptr[in, fs_options[vfat_options]], chdir bool8) fd_dir
 // fs_image_segment {
 //	data	ptr[in, array[int8]]
 //	size	len[data, intptr]
 //	offset	intptr
 // }
-static long syz_mount_image(volatile long fsarg, volatile long dir, volatile unsigned long size, volatile unsigned long nsegs, volatile long segments, volatile long flags, volatile long optsarg)
+static long syz_mount_image(volatile long fsarg, volatile long dir, volatile unsigned long size, volatile unsigned long nsegs, volatile long segments, volatile long flags, volatile long optsarg, volatile long change_dir)
 {
 	struct fs_image_segment* segs = (struct fs_image_segment*)segments;
-	int res = -1, err = 0, loopfd = -1, memfd = -1, need_loop_device = !!segs;
+	int res = -1, err = 0, loopfd = -1, memfd = -1, need_loop_device = !!segs, should_change_dir = !!change_dir;
```

Comment:
I don't see a reason to define the `should_change_dir` temporary. For need_loop_device it's done because it's used twice; using "segs" twice while being volatile generates worse code (and segs isn't the most obvious name for needing loop device).

But here you only refer to should_change_dir once, and "change_dir" is already a clear name, we can just use it directly below.